### ISSUE TITLE
Provide the scores [feature request]

### ISSUE THREAD

**Issue by cpbl** [jakevdp/wpca#5]
I came thinking that many users of PCA are looking to get new (a smaller set)  variables **Y** from a set of variables **X**.  But the example provided with this package does not appear to do that.  Moreover, the methods do not offer to do it for you.

In fact, to the clueless, it's not obvious how to even get these. And, moreover, the example that comes with this package does not tell you what to do if you have a set of non-commensurable variables **X**, ie which should be normalized according to standard practice, before running PCA.

I am not statistically sophisticated, and maybe there are reasons to make the above uses hard for people who would abuse PCA? Or maybe I have overlooked something. But, I believe all that needs to be added is:

- normalize the input X columns to make X~
- multiply the pca.components_.dot( X~)
- and make this available, eg as pca.scores_ ?

Could/should this be added to either the code or docs?

**Comment by jakevdp** [comment]
Is the ``transform`` method what you're looking for? For example, you can fit on a ``[1000, 20]`` matrix, and transform it to a ``[1000, 10]`` matrix. Is that what you mean by "a smaller set of variables"?

**Comment by cpbl** [comment]
Whoops, yes, it looks like it. Thank you!
 But I do not understand why that method should involve the weights again.  I'll come back to this and close/mark it as invalid when I've better understood that output.

**Comment by jakevdp** [comment]
The weights are required again because ``transform`` can be applied to a new dataset with new weights. If you want to fit and transform a single dataset, you can use the ``fit_transform`` method and then you only need to supply the data and the weights once.

**Comment by cpbl** [comment]
In my application I'm fitting on one dataset, using per-sample weights. That is, the weights vary from one row to another but not across columns. Thus I do not need to know any new weights when I use that fit to transform a different  data set. I believe your code accepts weights which differ even across variables for a given observation/row/sample, in which case one does need to know something about the weights even for `transform`.   If that all sounds right, you can close this, though I still think a second vignette could be useful for some who find your work.  Thanks!!

**Comment by jakevdp** [comment]
I'm a bit confused: if weights vary from one row to another, then when you apply it to a new dataset you need a new set of weights, right?

This package follows the scikit-learn convention where rows are samples, and columns are features.

**Comment by cpbl** [comment]
> This package follows the scikit-learn convention where rows are samples, and columns are features.

Good. 

I'm sorry for confusing you; I'm sure it's me. Thanks for explaining. I thought: `transform`ing is simply a linear combination across columns, within a row, isn't it? If so, and weights are constant within a row, I don't see what it means to weight rows.

**Comment by jakevdp** [comment]
What ``transform`` does is to project new data on the eigenvectors learned from the training data, taking into account weights associated with both the training data and the new data.

If you want to project the training data onto the eigenvectors computed from itself, then ``fit_transform`` does this in one step.
